src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -192,6 +222,36 @@ public AnalyzerConfigOptionsResult GetOptionsForSourcePath(string sourcePath)
                 var diagnosticBuilder = ArrayBuilder<Diagnostic>.GetInstance();
 
                 int sectionKeyIndex = 0;
+
+                if (_globalConfig is object)
+                {
+                    addOptions(_globalConfig.GlobalSection,
```

Comment:
You added the section here, but not above. This is broken.

Reply:
No, the above is just the 'section key' so it doesn't matter if we always add it or always don't the result is the same, we just use a little less memory in the cache this way.

---
In reply to: [425946268](https://github.com/dotnet/roslyn/pull/43889#discussion_r425946268) [](ancestors = 425946268)

Reply:
I think it's impossible to construct a test case for this right now because the AnalyzerConfig class is private, but what if you had two AnalyzerConfig files, which share the same named sections but a different global section?

Reply:
Yeah, but by this point we've merged the global section. So there can only ever be one?

---
In reply to: [425971242](https://github.com/dotnet/roslyn/pull/43889#discussion_r425971242) [](ancestors = 425971242)

Reply:
Ah you're right, the set is immutable at this point. My mistake!